lib/containers/common.pm

```diff
@@ -105,18 +99,24 @@ sub install_docker_when_needed {
     }
     systemctl('is-active docker');
     systemctl('status docker', timeout => 120);
-    assert_script_run('docker info');
+    record_info('docker', script_output('docker info'));
 }
 
 sub install_buildah_when_needed {
     my $host_os = shift;
-    my @pkgs = qw(buildah);
     if (script_run("which buildah") != 0) {
-        # We may run openSUSE with DISTRI=sle and opensuse doesn't have SUSEConnect
-        activate_containers_module if $host_os =~ 'sles';
-        zypper_call "in @pkgs";
-        record_info('buildah', script_output('buildah info'));
+        if ($host_os eq 'centos') {
+            assert_script_run "dnf -y update", timeout => 900;
```

Comment:
i am not sure but this will run twice in some cases, right? somewhere i have seen `install podman` and then `install buildah`. i wonder if we could do better here. But i dont think it is a blocker too.

Reply:
yeah, that's true, but it won't harm either :)